test/legacy/roles/one_image_facts/tasks/main.yml

```diff
@@ -0,0 +1,163 @@
+---
+- name: Fetch all images
+  one_image_facts:
+  register: all_images
+
+- name: Print all images
+  debug:
+    var: all_images
+
+- name: Gather facts about an image using a name
+  one_image_facts:
+    name: '{{ one_image_name }}'
+  register: image_with_name
+
+- name: Verify image's facts
+  assert:
+    that:
+      - not image_with_name is changed
+      - image_with_name.images|length == 1
+      - image_with_name.images[0].name == '{{ one_image_name }}'
+      - not image_with_name.images[0].used|bool
+      - image_with_name.images[0].running_vms == 0
+
+- name: Gather facts about the image using ID
+  one_image_facts:
+    ids: '{{ image_with_name.images[0].id }}'
+  register: image_with_ids
+
+- name: Verify image's facts
+  assert:
+    that:
+      - not image_with_ids is changed
+      - image_with_ids.images|length == 1
+      - image_with_ids.images[0].name == '{{ one_image_name }}'
+      - not image_with_ids.images[0].used|bool
+      - image_with_ids.images[0].running_vms == 0
+
+- name: Try to gather facts about an image using a name and ids
+  one_image_facts:
+    name: '{{ one_image_name }}'
+    ids: '{{ image_with_name.images[0].id }}'
+  register: image_name_ids
+  failed_when: not image_name_ids is failed
+
+- name: Try to fetch non-existent image by name
+  one_image_facts:
+    name: non-existent-vm-{{ ansible_date_time.iso8601_basic_short }}
+  register: image_missing
+  failed_when: not image_missing is failed
+
+- name: Try to gather facts about non-existent images by regex
+  one_image_facts:
+    name: ~non-existent-vm-{{ ansible_date_time.iso8601_basic_short }}-*
+  register: images_with_regex
+
+- name: Verify that images list is empty
+  assert:
+    that:
+      - not images_with_regex is changed
+      - images_with_regex.images|length == 0
+
+- name: Try to fetch non-existent image by id
+  one_image_facts:
+    id: -999
```

Comment:
hmm, seems `id` is already used in the tests, but aliases is missing?

Reply:
There was an idea for an alias, but it's typo.